pkg/instancegroups/instancegroups.go

```diff
@@ -468,3 +468,77 @@ func (g *CloudInstanceGroup) Delete(cloud fi.Cloud) error {
 
 	return nil
 }
+
+// ScaleInstanceGroup scale the cloud resources for an InstanceGroup
+type ScaleInstanceGroup struct {
+	Cluster         *api.Cluster
+	Cloud           fi.Cloud
+	DesiredReplicas int64
+
+	Clientset simple.Clientset
+}
+
+func (c *ScaleInstanceGroup) ScaleInstanceGroup(group *api.InstanceGroup) error {
+	groups, err := FindCloudInstanceGroups(c.Cloud, c.Cluster, []*api.InstanceGroup{group}, false, nil)
+	cig := groups[group.ObjectMeta.Name]
+
+	if cig == nil {
+		return fmt.Errorf("AutoScalingGroup %q not found in cloud - skipping scaling", group.ObjectMeta.Name)
+	}
+
+	if len(groups) != 1 {
+		return fmt.Errorf("Multiple InstanceGroup resources found in cloud")
+	}
+
+	cig.asg.DesiredCapacity = &c.DesiredReplicas
+	glog.Infof("Scaling InstanceGroup %q instances size to: %v", group.ObjectMeta.Name, c.DesiredReplicas)
+
+	// decrease max cluster size
+	if c.DesiredReplicas < *cig.asg.MinSize {
+		glog.Infof("Changing min instances from %v to %v", *cig.asg.MinSize, c.DesiredReplicas)
+		cig.asg.MinSize = &c.DesiredReplicas
+		group.Spec.MinSize = fi.Int32(int32(c.DesiredReplicas))
+	}
+
+	// increasing max cluster size
+	if c.DesiredReplicas > *cig.asg.MaxSize {
+		glog.Infof("Changing max instances from %v to %v", *cig.asg.MaxSize, c.DesiredReplicas)
+		cig.asg.MaxSize = &c.DesiredReplicas
+		group.Spec.MaxSize = fi.Int32(int32(c.DesiredReplicas))
+	}
+
+	err = cig.Scale(c.Cloud)
+	if err != nil {
+		return fmt.Errorf("error scaling InstanceGroup: %v", err)
+	}
+
+	_, err = c.Clientset.InstanceGroupsFor(c.Cluster).Update(group)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (g *CloudInstanceGroup) Scale(cloud fi.Cloud) error {
+
+	c := cloud.(awsup.AWSCloud)
```

Comment:
I worry we might have broken this with some of the refactorings of this interface that have gone in recently... it should be easier now, but let me know if you need some pointers figuring out where things have moved to!